src/system_database.ts

```diff
@@ -895,11 +1021,30 @@ export class PostgresSystemDatabase implements SystemDatabase {
     await this.recordWorkflowStatusChange(workflowID, status, { resetRecoveryAttempts });
   }
 
+  setWFCancelMap(workflowID: string) {
```

Comment:
I couldn't find any read of the `workflow_cancel` table -- what happens if the DB notifier becomes slow or unreliable?